### PR TITLE
Add some components and some functionality

### PR DESCRIPTION
**Pull request by Dhaiwat10**
Components added:

- Totals (statistics at the top)
- Header
- Legend
- LanguagePicker

Functionality added:

- Fetch totals (statistics at the top)
- Automatically detect dark/light mode through the user's system preference

Functionalities WIP:

- Integrate the Tab menu
- Add color coding to the map
- Add the time slider and the date picker

### CODE REVIEW

**Comment by Dhaiwat10** [comment]
Issues covered: #187, #186, #178.

**Comment by Dhaiwat10** [comment]
More updates:

- Fix styling of the legend component
- Hook up the tab menu on the right
- Add logic for map data fetching from the old API (v1)
- Add color coding to the world map

**Comment by jeff-knurek** [comment]
@Dhaiwat10 I can't quite figure out why these files are showing as changing:
```
 .github/workflows/azure-static-web-apps-polite-cliff-0e188b310.yml 
 DB/sample-data/LayoutSchema.json 
 DB/sample-data/SnapshotSchema1.json 
```
maybe it's something with your local env or editor?? can you try to revert them back? or if they should be changed, maybe create a different PR just for what the mystery change is :pray:

File: MAP/.gitignore [hunk 1]
```diff
@@ -24,4 +24,7 @@ yarn-error.log*
 
 # IDE
 .idea
-.eslintcache
\ No newline at end of file
+
+.eslintcache
```
Comment (jeff-knurek) [inline MAP/.gitignore:28]: :+1:

**Comment by Dhaiwat10** [comment]
@jeff-knurek sure thing. I'll look into what went wrong with these files. Though from what I can see, it seems like a silly issue with my local IDE. Will fix it and get back to you.

File: MAP/src/App.test.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+// import { render, screen } from '@testing-library/react';
```
Comment (jeff-knurek) [inline MAP/src/App.test.js:1]: it seems like commenting out a test isn't the best strategy _(even if it isn't useful now)_

**Comment by Dhaiwat10** [comment]
I copied the contents of all three files from the master branch codebase. They still show up as changed, even though the content is literally the same and the lines deleted are the same as the lines added. Tricky.

File: MAP/src/App.test.js (hunk 2 above)
Comment (Dhaiwat10) [reply MAP/src/App.test.js:1]: The only test was the default test case that comes in-built with Create React App. It expects a 'Learn React' link to be rendered on the screen, which of course would not pass. Do still want us to include the test file?

Comment (jeff-knurek) [reply MAP/src/App.test.js:1]: yeah, I saw the test you commented out wasn't a good one, but I wonder if there is something else that renders on the screen that could be tested for instead?? maybe if we really are going to remove the test and not add something new, it's better to just delete the file :shrug:

File: MAP/src/components/LanguageSelector/LanguageSelector.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          {world}
+        </div>
+      </div>
+      <div
+        className={`${classes.LangOptions} ${showLanguages && classes.show}`}
+      >
+        {languages.map((language) => {
+          return (
+            <div
+              style={{ color: `${dark ? 'white' : '#333333'}` }}
```
Comment (filipkowal) [inline MAP/src/components/LanguageSelector/LanguageSelector.js:32]: Warning: "Each child in a list should have a unique 'key' prop."

**Review by filipkowal — COMMENTED** [review]
There are some new errors and warnings in storybook components. 

image
image
image

**Comment by Dhaiwat10** [comment]
Noted your pointers. Thanks for the review!

File: MAP/src/components/LanguageSelector/LanguageSelector.js (hunk 3 above)
Comment (Dhaiwat10) [reply MAP/src/components/LanguageSelector/LanguageSelector.js:32]: Creating an issue for these Storybook problems.

**Comment by Dhaiwat10** [comment]
> There are some new errors and warnings in storybook components.
> 
> image
> image
> image

Creating an issue for these Storybook problems.

**Comment by JFQueralt** [comment]
I may have forgotten to mention that the Lockdown header is to be removed. It has no purpose and is redundant with the banner on the website once embedded.

**Comment by Dhaiwat10** [comment]
Merging the PR since all checks are passing.

**Pull request merged**